src/plugins/intel_npu/tests/functional/behavior/npu_driver_compiler_adapter/custom_stream.cpp

```diff
@@ -0,0 +1,128 @@
+        auto add_constant = ov::opset11::Constant::create(ov::element::f16, ov::Shape{1}, {0.5});
+        auto add = std::make_shared<ov::opset11::Add>(mul, add_constant);
+        // Just a sample model here, large iteration to make the model large
+        for (int i = 0; i < 1000; i++) {
+            add = std::make_shared<ov::opset11::Add>(add, add_constant);
+        }
+        auto res = std::make_shared<ov::opset11::Result>(add);
+
+        /// Create the OpenVINO model
+        return std::make_shared<ov::Model>(ov::ResultVector{std::move(res)}, ov::ParameterVector{std::move(data)});
+    }
+
+    size_t getFileSize(std::istream& strm) {
+        const size_t streamStart = strm.tellg();
+        strm.seekg(0, std::ios_base::end);
+        const size_t streamEnd = strm.tellg();
+        const size_t bytesAvailable = streamEnd - streamStart;
+        strm.seekg(streamStart, std::ios_base::beg);
+        return bytesAvailable;
+    }
+
+    void SetUp() override {
+        std::tie(target_device, configuration) = this->GetParam();
+        SKIP_IF_CURRENT_TEST_IS_DISABLED()
+        OVPluginTestBase::SetUp();
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<CompilationParams> obj) {
+        std::string targetDevice;
+        ov::AnyMap configuration;
+        std::tie(targetDevice, configuration) = obj.param;
+        std::replace(targetDevice.begin(), targetDevice.end(), ':', '.');
+
+        std::ostringstream result;
+        result << "targetDevice=" << targetDevice << "_";
+        result << "targetPlatform=" << ov::test::utils::getTestsPlatformFromEnvironmentOr(targetDevice) << "_";
+        if (!configuration.empty()) {
+            for (auto& configItem : configuration) {
+                result << "configItem=" << configItem.first << "_";
+                configItem.second.print(result);
+            }
+        }
+        return result.str();
+    }
+
+    void TearDown() override {
+        if (!configuration.empty()) {
+            utils::PluginCache::get().reset();
+        }
+        APIBaseTest::TearDown();
+    }
+
+protected:
+    ov::AnyMap configuration;
+};
+
+TEST_P(DriverCompilerAdapterCustomStreamTestNPU, TestLargeModel) {
+    auto model = createModelWithLargeSize();
+    IR irModel(model, 11, true);
+    std::istream& xmlStream = irModel.getXml();
+    std::istream& weightsStream = irModel.getWeights();
+    size_t xmlSize = getFileSize(xmlStream);
+    size_t weightsSize = getFileSize(weightsStream);
+    {
+        std::vector<char> xml(xmlSize);
+        xmlStream.read(xml.data(), xmlSize);
+        std::ofstream xmlFile("./model-dump.xml", std::ios::binary);
+        if (xmlFile) {
+            xmlFile.write(xml.data(), xmlSize);
+            xmlFile.close();
+        }
+
+        std::vector<char> weights(weightsSize);
+        weightsStream.read(weights.data(), weightsSize);
+        std::ofstream binFile("./model-dump.bin", std::ios::binary);
```

Comment:
Should we use random name? otherwise hard to ensure it's not some leftover file perhaps

Reply:
Use random name from time seed now